mockwebserver/src/main/kotlin/okhttp3/mockwebserver/MockWebServer.kt

```diff
@@ -17,1144 +17,53 @@
-      message = "moved to val",
-      replaceWith = ReplaceWith(expression = "requestCount"),
-      level = DeprecationLevel.ERROR)
-  fun getRequestCount(): Int = requestCount
-
-  /**
-   * Scripts [response] to be returned to a request made in sequence. The first request is
-   * served by the first enqueued response; the second request by the second enqueued response; and
-   * so on.
-   *
-   * @throws ClassCastException if the default dispatcher has been
-   * replaced with [setDispatcher][dispatcher].
-   */
-  fun enqueue(response: MockResponse) =
-      (dispatcher as QueueDispatcher).enqueueResponse(response.clone())
-
-  /**
-   * Starts the server on the loopback interface for the given port.
-   *
-   * @param port the port to listen to, or 0 for any available port. Automated tests should always
-   * use port 0 to avoid flakiness when a specific port is unavailable.
-   */
-  @Throws(IOException::class)
-  @JvmOverloads fun start(port: Int = 0) = start(InetAddress.getByName("localhost"), port)
-
-  /**
-   * Starts the server on the given address and port.
-   *
-   * @param inetAddress the address to create the server socket on
-   * @param port the port to listen to, or 0 for any available port. Automated tests should always
-   * use port 0 to avoid flakiness when a specific port is unavailable.
-   */
-  @Throws(IOException::class)
-  fun start(inetAddress: InetAddress, port: Int) = start(InetSocketAddress(inetAddress, port))
-
-  /**
-   * Starts the server and binds to the given socket address.
-   *
-   * @param inetSocketAddress the socket address to bind the server on
-   */
-  @Synchronized @Throws(IOException::class)
-  private fun start(inetSocketAddress: InetSocketAddress) {
-    require(!started) { "start() already called" }
-    started = true
-
-    this.inetSocketAddress = inetSocketAddress
-
-    serverSocket = serverSocketFactory!!.createServerSocket()
-
-    // Reuse if the user specified a port
-    serverSocket!!.reuseAddress = inetSocketAddress.port != 0
-    serverSocket!!.bind(inetSocketAddress, 50)
-
-    portField = serverSocket!!.localPort
-
-    taskRunner.newQueue().execute("MockWebServer $portField", cancelable = false) {
-      try {
-        logger.fine("$this starting to accept connections")
-        acceptConnections()
-      } catch (e: Throwable) {
-        logger.log(Level.WARNING, "$this failed unexpectedly", e)
-      }
-
-      // Release all sockets and all threads, even if any close fails.
-      serverSocket?.closeQuietly()
-
-      val openClientSocket = openClientSockets.iterator()
-      while (openClientSocket.hasNext()) {
-        openClientSocket.next().closeQuietly()
-        openClientSocket.remove()
-      }
-
-      val httpConnection = openConnections.iterator()
-      while (httpConnection.hasNext()) {
-        httpConnection.next().closeQuietly()
-        httpConnection.remove()
-      }
-      dispatcher.shutdown()
-    }
+class MockWebServer : TestRule, MockWebServer() {
```

Comment:
nice.

Reply:
wanna delegate instead of subclassing? It’s a bit more boilerplate but then we don’t need to open ’er up.